apps/gk_species_radiation.c

```diff
@@ -166,6 +182,18 @@ void
 gk_species_radiation_moms(gkyl_gyrokinetic_app *app, const struct gk_species *species,
   struct gk_rad_drag *rad, const struct gkyl_array *fin[], const struct gkyl_array *fin_neut[])
 {
+  // compute needed moments
+  gk_species_moment_calc(&rad->lab_moms, species->local, app->local, species->f);
+  if (app->use_gpu) {
+    gkyl_mom_calc_bcorr_advance_cu(rad->bcorr_calc, &species->local, &app->local, species->f, rad->boundary_corrections);
+    gkyl_prim_lbo_calc_advance_cu(rad->coll_pcalc, &app->local, rad->lab_moms.marr, rad->boundary_corrections, rad->prim_moms);
+  }
+  else {
+    gkyl_mom_calc_bcorr_advance(rad->bcorr_calc, &species->local, &app->local, species->f, rad->boundary_corrections);
+    gkyl_prim_lbo_calc_advance(rad->coll_pcalc, &app->local, rad->lab_moms.marr, rad->boundary_corrections, rad->prim_moms);
+  }
+  gkyl_array_set_offset(rad->vtsq, 1.0, rad->prim_moms, 1*app->confBasis.num_basis);
+
```

Comment:
You can also just use MaxwellianMoments in the same way we do for gk_species_bgk (making a gk_species_moment of type MaxwellianMoments) and that will give you T/m.

Reply:
but that would return m0 and upar, which are not needed here, so it seems `dg_prim_vars_gyrokinetic_new` is simpler and I thought that was its purpose

Reply:
You need m0 in general (since these are the electron moments and you have to get ne), and you always have to compute upar to get T/m (or if you prefer you always have to do some weak division by M0) so it is sort of irrelevant cost-wise.

Reply:
Currently, the `gk_species_radiation_moment` (the function where vtsq is being calculated) doesn't calculate the M0 moment, but it's calculated in this function when the electron density dependence is included. The `dg_prim_varys_gyrokinetic` doesn't seem to have an advance method and if I were to directly call the kernels it would be called multiple times per cell since it should be in `gkyl_dg_calc_gk_rad_vars_nI_nu_advance` I believe. So, the MaxwellianMoments seems like a better idea here to me.